src/Symfony/Bridge/Twig/Extension/FormExtension.php

```diff
@@ -166,6 +167,22 @@ public function isRootForm(FormView $formView)
         return null === $formView->parent;
     }
 
+    /**
+     * @internal
+     */
+    public function encodeCurrency(Environment $environment, $text, $widget = '')
+    {
+        if ('UTF-8' === $charset = $environment->getCharset()) {
+            $text = htmlspecialchars($text, ENT_QUOTES | (\defined('ENT_SUBSTITUTE') ? ENT_SUBSTITUTE : 0), 'UTF-8');
+        } else {
+            $text = htmlentities($text, ENT_QUOTES | (\defined('ENT_SUBSTITUTE') ? ENT_SUBSTITUTE : 0), 'UTF-8');
+            $text = iconv('UTF-8', $charset, $text);
+            $widget = iconv('UTF-8', $charset, $widget);
```

Comment:
Are we sure $widget is always in UTF-8? This looks strange to me, because that would mean this is also bugged. Is it?

Reply:
i think we can assume we get raw html in UTF from the form component. Here we include it in a new safe html value concerning the target charset, using `htmlentities` instead of `htmlspecialchars` for escaping. 

In general we are consistent regular escaped output (also in target charset) from `twig_escape_filter`.

Reply:
At least we treated this as UTF before also (result from `block()` in twig).